doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -700,57 +648,30 @@ the conversion in hypotheses :n:`{+ @ident}`.
-
-      Replaces :n:`@qualid` in hypothesis (or hypotheses) designated
-      by :token:`goal_occurrences` with its definition and replaces
-      the hypothesis with its :math:`\beta`:math:`\iota` normal form.
-
-   .. tacv:: unfold {+, @qualid}
-
-      Replaces :n:`{+, @qualid}` with their definitions and replaces
-      the current goal with its :math:`\beta`:math:`\iota` normal
-      form.
-
-   .. tacv:: unfold {+, @qualid at @occurrences }
-
-      The list :token:`occurrences` specify the occurrences of
-      :n:`@qualid` to be unfolded. Occurrences are located from left
-      to right.
-
       .. exn:: Bad occurrence number of @qualid.
          :undocumented:
 
       .. exn:: @qualid does not occur.
          :undocumented:
 
-   .. tacv:: unfold @string
-
-      If :n:`@string` denotes the discriminating symbol of a notation
-      (e.g. "+") or an expression defining a notation (e.g. `"_ +
-      _"`), and this notation denotes an application whose head symbol
-      is an unfoldable constant, then the tactic unfolds it.
-
-   .. tacv:: unfold @string%@ident
+.. tacn:: fold {+ @one_term } @simple_occurrences
 
-      This is variant of :n:`unfold @string` where :n:`@string` gets
-      its interpretation from the scope bound to the delimiting key
-      :token:`ident` instead of its default interpretation (see
-      :ref:`Localinterpretationrulesfornotations`).
-
-   .. tacv:: unfold {+, {| @qualid | @string{? %@ident } } {? at @occurrences } } {? in @goal_occurrences }
-
-      This is the most general form.
+   First, this tactic reduces each :n:`@one_term` using the :tacn:`red` tactic.
+   Then, every occurrence of the resulting terms in the selected hypotheses and/or
+   goal will be replaced by its associated :n:`@one_term`. This tactic is particularly
+   useful for
+   reversing undesired unfoldings, which may make the goal very hard to read.
+   The undesired unfoldings may be due to the limited capabilities of
+   other reduction tactics.
+   On the other hand, when an unfolded function applied to its argument has been
+   reduced, the :tacn:`fold` tactic doesn't do anything.
 
-.. tacn:: fold @term
-   :name: fold
+   :tacn:`fold` :n:`@one_term__1 @one_term__2` is equivalent to
+   :n:`fold @one_term__1; fold @one_term__2`.
 
-   This tactic applies to any goal. The term :n:`@term` is reduced using the
-   :tacn:`red` tactic. Every occurrence of the resulting :n:`@term` in the goal is
-   then replaced by :n:`@term`. This tactic is particularly useful when a fixpoint
-   definition has been wrongfully unfolded, making the goal very hard to read.
-   On the other hand, when an unfolded function applied to its argument has been
-   reduced, the :tacn:`fold` tactic won't do anything.
+   .. todo PR: I don't entirely follow the first example or gather what
+      points it's trying to show.  What's with the "Fail progress" and
+      "pattern"?  At the least it could use some explanation.  Théo?
```

Comment:
Yes, sorry about that. The `Fail progress fold not` is just showing that `fold not` doesn't do anything. So it would be better to replace it by a simple call to `fold not` + a textual description "as you can see `fold not` doesn't work in this case". The sentence above "On the other hand, when an unfolded function applied to its argument has been reduced, the :tacn:`fold` tactic doesn't do anything." isn't really part of the tactic description so it could be moved as the introduction of this example. Also the title of the example could be "`fold` is not always sufficient to undo `unfold`".